components/locale/examples/filter_langids.rs

```diff
@@ -0,0 +1,54 @@
+// A sample application which takes a comma separated list of language identifiers,
+// filters out identifiers with language subtags different than `en` and serializes
+// the list back into a comma separated list in canonical syntax.
+use std::env;
+
+use icu_locale::{subtags, LanguageIdentifier};
+
+const DEFAULT_INPUT: &str =
+    "de, en-us, zh-hant, sr-cyrl, fr-ca, es-cl, pl, en-latn-us, ca-valencia, und-arab";
+
+fn filter_input(input: &str) -> String {
+    // 1. Parse the input string into a list of language identifiers.
+    let langids: Vec<LanguageIdentifier> = input
+        .split(',')
+        .filter_map(|s| s.trim().parse().ok())
+        .collect();
+
+    // 2. Filter for LanguageIdentifiers with Language subtag `en`.
+    let en_lang: subtags::Language = "en".parse().expect("Failed to parse language subtag.");
```

Comment:
Is this a good place to demo the macros?

Reply:
once we land them, I'll switch to them in examples :)